apps/services/endorsements/api/src/app/modules/endorsement/endorsement.service.ts

```diff
@@ -68,6 +80,33 @@ export class EndorsementService {
       orderOption: [['counter', 'DESC']],
       where: { endorsementListId: listId },
     })
+
+    // endorsement data display rules for everyone(public,users,admins) except owner ...
+    // ... list owner sees all data
+    const user = this.request.auth as User
+    const listOwnerNationalId = await this.getListOwnerNationalId(listId)
+    if (user?.nationalId != listOwnerNationalId) {
```

Comment:
Should this be in the interceptor files, where we are already masking nationalIds?

Reply:
good call - I move it to the interceptor